pkg/modules/generators/trait/ops_rule_generator.go

```diff
@@ -1,36 +1,54 @@
 package trait
 
 import (
+	"fmt"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/intstr"
 	"kusionstack.io/kube-api/apps/v1alpha1"
 
 	"kusionstack.io/kusion/pkg/apis/intent"
 	"kusionstack.io/kusion/pkg/apis/project"
 	"kusionstack.io/kusion/pkg/apis/stack"
+	workspaceapi "kusionstack.io/kusion/pkg/apis/workspace"
 	"kusionstack.io/kusion/pkg/modules"
 	appmodule "kusionstack.io/kusion/pkg/modules/inputs"
+	"kusionstack.io/kusion/pkg/modules/inputs/trait"
 	"kusionstack.io/kusion/pkg/modules/inputs/workload"
+	"kusionstack.io/kusion/pkg/workspace"
 )
 
 type opsRuleGenerator struct {
-	project *project.Project
-	stack   *stack.Stack
-	appName string
-	app     *appmodule.AppConfiguration
+	project         *project.Project
+	stack           *stack.Stack
+	appName         string
+	app             *appmodule.AppConfiguration
+	workspaceConfig map[string]workspaceapi.GenericConfig
 }
 
 func NewOpsRuleGenerator(
 	project *project.Project,
 	stack *stack.Stack,
 	appName string,
 	app *appmodule.AppConfiguration,
+	ws *workspaceapi.Workspace,
 ) (modules.Generator, error) {
+	if project == nil || len(project.Name) == 0 {
+		return nil, fmt.Errorf("project name must not be empty")
+	}
+	if ws == nil {
+		return nil, fmt.Errorf("workspace must not be empty")
+	}
+	wsConfig, err := workspace.GetProjectModuleConfigs(ws.Modules, project.Name)
```

Comment:
You can get moduleConfigs in `pkg/modules/generators/app_configurations_generator.go`, and only use the opsRule config in workspace as input.
[PR#709](https://github.com/KusionStack/kusion/pull/709) does so.